TestProject/ClientApp/src/components/UserProfile.js

```diff
@@ -1,16 +1,47 @@
-﻿import React from 'react';
+﻿import React, { useState, useEffect } from 'react';
+import { NotificationContainer, NotificationManager } from 'react-notifications';
+import 'react-notifications/lib/notifications.css';
 import Posts from './Posts';
 
-function UserProfile({ userProfile }) {
-    return <div>
-        <div>Name: {userProfile.name}</div>
-        <div>Email: {userProfile.email}</div>
-        <div>Rating: {userProfile.rating}</div>
-        {userProfile.id
-            && <Posts userLoggedIn={true}
-                userProfile={userProfile}
-                filterByCurrentUser={true} />}
-    </div>;
+function UserProfile({ currentUser, userId }) {
```

Comment:
The component shouldn't decide what page to render depending on availability of the Id. It feels unexpected, that it renders current user by default.